### PR TITLE
Feature: Add a cli command to support create new project.

### PR DESCRIPTION
**Pull request by chaosddp**
# Description

Add a cli command to create a project under current working folder, it only contains Environment parts, without any algorithms.

It supports 2 modes:

1. Create a runner.py that use built-in scenario with/without built-in topology.
2. Create a customized scenario from scratch, with a runnable runner.py.

Usage:

The new command is:

```sh
maro project new
```

NOTE: this command will generate files and folders in current folder, so usually users should create a new project folder first.

Then it will show a step by step wizard to ask users what they want :

1. Use built-in scenarios.

```sh
Use built-in scenario?yes
Scenario name:cim
Use built-in topology (configuration)?yes
New topology name (configuration) to use:global_trade.22p_l0.0
Durations to emulate:1024
Number of episodes to emulate:500
{'durations': 1024,
 'scenario': 'cim',
 'topology': 'global_trade.22p_l0.0',
 'total_episodes': 500,
 'use_builtin_scenario': True,
 'use_builtin_topology': True}

Is this OK?yes

```

This will create a runner.py to interactive with environment, and may contains a topologies folder that contains customized configuration that copied from built-in one.

2. Create a customized scenario, use current folder name as new scenario name by default.

```sh
Use built-in scenario?no
New scenario name:my_test
New topology name:my_test
Durations to emulate:1000
Number of episodes to emulate:100
{'durations': 1000,
 'scenario': 'my_test',
 'topology': 'my_test',
 'total_episodes': 100,
 'use_builtin_scenario': False,
 'use_builtin_topology': False}

Is this OK?yes

```

The result folder structure will like:

```
-- runner.py
-- scenario
   -- business_engine.py
   -- common.py
   -- events.py
   -- frame_builder.py
   -- topologies
      -- my_test
         -- config.yml
```

## Linked issue(s)/Pull request(s)

- [issue_number](issue_link)

## Type of Change

- [ ] Non-breaking bug fix
- [ ] Breaking bug fix
- [x] New feature
- [ ] Test
- [ ] Doc update
- [ ] Docker update

## Related Component

- [x] Simulation toolkit
- [ ] RL toolkit
- [ ] Distributed toolkit

## Has Been Tested

- OS:
  - [x] Windows
  - [x] Mac OS
  - [x] Linux
- Python version:
  - [x] 3.6
  - [x] 3.7
- Key information snapshot(s):

## Needs Follow Up Actions

- [ ] New release package
- [ ] New docker image

## Checklist

- [ ] Add/update the related comments
- [ ] Add/update the related tests
- [ ] Add/update the related documentations
- [ ] Update the dependent downstream modules usage

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -3,3 +3,4 @@ prune examples
 
 include maro/simulator/scenarios/cim/topologies/*/*.yml
 include maro/simulator/scenarios/citi_bike/topologies/*/*.yml
```
[the remark this answers is not in the document]
Comment (chaosddp) [reply MANIFEST.in:5]: yes, seems like the topologies not included in package, will update it

File: maro/cli/project_generator/project_generator.py [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+    def __init__(self, err_msg: str):
+        super().__init__()
+        self._err_msg = err_msg
+
+    def validate(self, document):
+        if not document.text or document.text.strip() == "":
+            raise ValidationError(message=self._err_msg)
+
+
+class NumberValidator(Validator):
```
[the remark this answers is not in the document]
Comment (chaosddp) [reply maro/cli/project_generator/project_generator.py]: fixed

File: maro/cli/maro.py [hunk 3]
```diff
@@ -995,6 +1004,22 @@ def load_parser_inspector(prev_parser: ArgumentParser, global_parser: ArgumentPa
...
 
 
+def load_parser_project(prev_parser: ArgumentParser, global_parser: ArgumentParser):
+    sub_parsers = prev_parser.add_subparsers()
+
+    from maro.cli.project_generator.project_generator import new_project
+
+    # maro project new
+    new_cmd_parser = sub_parsers.add_parser(
+        "new",
```
[the remark this answers is not in the document]
Comment (chaosddp) [reply maro/cli/maro.py:1014]: Ok, just found that we have no a section about cli commands.

File: MANIFEST.in (hunk 1 above)
[the remark this answers is not in the document]
Comment (chaosddp) [reply MANIFEST.in:5]: fixed

File: maro/cli/maro.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (chaosddp) [reply maro/cli/maro.py:1014]: Added "scenarios/commandline.rst" with a brief intro about new command.

File: docs/source/index.rst [hunk 4]
```diff
@@ -93,6 +93,7 @@ Contents
     scenarios/container_inventory_management.rst
     scenarios/citi_bike.rst
     scenarios/vm_scheduling.rst
+    scenarios/commandline.rst
```
Comment (Jinyu-W) [inline docs/source/index.rst:96]: rename to `command_line.rst`?

File: docs/source/scenarios/commandline.rst [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    Is this OK?yes
+
+If these settings correct, then this command will create a runner.py script, you can just run with:
+
+.. code-block:: sh
+
+    python runner.py
+
+This script contains minimize code to interactive with environment without any action, you can then extend it as you wish.
```
Comment (Jinyu-W) [inline docs/source/scenarios/commandline.rst:45]: minimal code?

File: docs/source/scenarios/commandline.rst [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+Also you can create you own topology (configuration) if you say "no" for options "Use built-in topology (configuration)?".
+It will ask you for a name of new topology, then copy the content from built-in one into your working folder (topologies/your_topology_name/config.yml).
+
+
+2. Create a customized scenario
+-------------------------------
+
+This mode is used to generate a template of customize scenario for you instead of write it from scratch.
```
Comment (Jinyu-W) [inline docs/source/scenarios/commandline.rst:54]: customized scenario?

Comment (Jinyu-W) [inline docs/source/scenarios/commandline.rst:54]: instead of writing

File: docs/source/scenarios/commandline.rst [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    'use_builtin_topology': False}
+
+    Is this OK?yes
+
+This will generate following files like below:
+
+.. code-block:: sh
+
+    -- runner.py
+    -- scenario
```
Comment (Jinyu-W) [inline docs/source/scenarios/commandline.rst:78]: should the level of `scenario` higher than the py files?

File: maro/cli/project_generator/templates/frame_builder.py.jinja [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    self._age = age
+    self._capacity = capacity
+    self._inventory = inventory
+
+    self.reset()
+
+  def reset(self):
+    # Reset attributes with initial value.
+    self.age = self._age
+    self._capacity = self._capacity
```
Comment (Jinyu-W) [inline maro/cli/project_generator/templates/frame_builder.py.jinja:29]:
```
self.capacity = self._capacity
self.inventory = self._inventory
```
？

File: docs/source/scenarios/commandline.rst (hunk 7 above)
Comment (chaosddp) [reply docs/source/scenarios/commandline.rst:78]: No, this is the folder name that same level as runner.py

File: maro/cli/project_generator/templates/frame_builder.py.jinja (hunk 8 above)
Comment (chaosddp) [reply maro/cli/project_generator/templates/frame_builder.py.jinja:29]: fixed

File: docs/source/scenarios/commandline.rst (hunk 6 above)
Comment (chaosddp) [reply docs/source/scenarios/commandline.rst:54]: fixed

Comment (chaosddp) [reply docs/source/scenarios/commandline.rst:54]: fixed

File: docs/source/scenarios/commandline.rst (hunk 5 above)
Comment (chaosddp) [reply docs/source/scenarios/commandline.rst:45]: fixed

File: docs/source/index.rst (hunk 4 above)
Comment (chaosddp) [reply docs/source/index.rst:96]: fixed

File: docs/source/scenarios/commandline.rst (hunk 7 above)
Comment (Jinyu-W) [reply docs/source/scenarios/commandline.rst]: same for `business_engine.py`, `common.py`?

Comment (chaosddp) [reply docs/source/scenarios/commandline.rst]: no, scenario related files were under scenario folder, runner.py is the only script under the root.

Comment (Jinyu-W) [reply docs/source/scenarios/commandline.rst]: then the indents of line 79 ~ 85 should be extended

Comment (chaosddp) [reply docs/source/scenarios/commandline.rst]: fixed

**Review by Jinyu-W — APPROVED** [review]

**Pull request merged**